From 2e42a689dfec26b481f6314c5c027c89bc896087 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:48 -0400
Subject: [infiniband] cxgb4: fix for 64-bit integer division

Message-id: <1410475447-94392-90-git-send-email-svemuri@redhat.com>
Patchwork-id: 93820
O-Subject: [RHEL7.1 PATCH BZ 1124947 089/108] iw_cxgb4: fix for 64-bit integer division
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Fixed error introduced in commit id 7730b4c (" cxgb4/iw_cxgb4: work request
logging feature") while compiling on 32 bit architecture reported by kbuild.

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit da388973d4a15e71cada1219d625b5393c90e5ae)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/device.c b/drivers/infiniband/hw/cxgb4/device.c
index 9733343..74191c3 100644
--- a/drivers/infiniband/hw/cxgb4/device.c
+++ b/drivers/infiniband/hw/cxgb4/device.c
@@ -33,6 +33,7 @@
 #include <linux/moduleparam.h>
 #include <linux/debugfs.h>
 #include <linux/vmalloc.h>
+#include <linux/math64.h>
 
 #include <rdma/ib_verbs.h>
 
@@ -150,7 +151,7 @@ static int wr_log_show(struct seq_file *seq, void *v)
  int prev_ts_set = 0;
  int idx, end;
 
-#define ts2ns(ts) ((ts) * dev->rdev.lldi.cclk_ps / 1000)
+#define ts2ns(ts) div64_ul((ts) * dev->rdev.lldi.cclk_ps, 1000)
 
  idx = atomic_read(&dev->rdev.wr_log_idx) &
   (dev->rdev.wr_log_size - 1);
-- 
1.7.1